doc/design/embargo-1.md

```diff
@@ -0,0 +1,95 @@
+# Embargo support
+
+## Principles
+
+### Use of key stores
+
+Rely on the fact that "stores" (`blobs/`, and future `zarr/` see #295) although "public" are not pragmatically "usable" for users, since the assets named based on their IDs, and provide no semantic meaning (not even a file extension).
+It would be an expensive procedure for someone to "mine" the store(s) for some relevant data they would like to "steal".
+
+### "Move" operation within/between buckets is "expensive"
+
+There is no AWS S3 API for quick move (renaming) of a key within a bucket.
+As a result, "move" operation can be **very** expensive in terms of traffic (thus possibly cost if not "sponsored"), time it would take for transfer, implementation e.g. handling of possible errors during transfer, etc
+
+### Publishing a dandiset is enabled only for Open dandisets
+
+### Use-cases to fulfill
+
+- UC-Collaboration: authorized users of emabrgoed dataset should be able to manipulate it as easily as if it was a public dataset.
+- UC-Review: ability to make embargoed dataset available to other users "read-only".
```

Comment:
I agree that we should enable read-only access to datasets. I don't know if I would call it "Review." I think this could be also used in a user-case where one lab is collecting data and using DANDI to share it with a theorist in another lab.
```suggestion
- UC-Read-only: ability to make embargoed dandiset available to other users "read-only".
```

Reply:
A nice valid use case.  I just just trying to come up with Use Case "codenames" which are depicting the purpose (for your case would be something like `UC-Consumer` more than a "mechanism" (read-only).  I am not married to those names though, so can accept your suggestion or we could extend it to
```suggestion
- UC-Review, UC-Consumer: ability to make embargoed dataset available to other users "read-only".
```
or alike: